source/common/rds/route_config_provider_manager.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "envoy/rds/route_config_provider.h"
+
+namespace Envoy {
+namespace Rds {
+
+/**
+ * The RouteConfigProviderManager interface exposes helper functions for
+ * RouteConfigProvider implementations
+ */
+class RouteConfigProviderManager {
+public:
+  virtual ~RouteConfigProviderManager() = default;
+
+  virtual void eraseStaticProvider(RouteConfigProvider* provider) PURE;
```

Comment:
Actually yes, any value is valid here. The pointer will be used just for lookup in a set.
It could be const but in the original implementation the set contained non-const pointers [](https://github.com/envoyproxy/envoy/blob/0b53a6bb612e197ada5b04931a0b7ddd4cab0dd9/source/common/router/rds_impl.h#L267).